freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/common/DetailedStackStatus.java

```diff
@@ -78,7 +78,8 @@ public enum DetailedStackStatus {
     // CCM upgrade statuses
     UPGRADE_CCM_REQUESTED(Status.UPGRADE_CCM_REQUESTED, AvailabilityStatus.AVAILABLE),
     UPGRADE_CCM_IN_PROGRESS(Status.UPGRADE_CCM_IN_PROGRESS, AvailabilityStatus.AVAILABLE),
-    UPGRADE_CCM_FAILED(Status.UPGRADE_CCM_FAILED, AvailabilityStatus.UNAVAILABLE);
+    UPGRADE_CCM_FAILED(Status.UPGRADE_CCM_FAILED, AvailabilityStatus.UNAVAILABLE),
+    UPGRADE_FAILED(Status.UPGRADE_FAILED, AvailabilityStatus.UNAVAILABLE);
```

Comment:
this shouldn't be unavailable, as FreeIPA would be available. by marking it unavailable it would prevent provisioning new clusters etc. this is not the case, as freeipa is just working fine, maybe one instance is not provisioned well, but the others can still handle the requests

Reply:
I believe you, but if something error happens during the AWS migration and it will cause any problem (user loss, data loss or smtgh), I will address the ticket to you

Reply:
feel free, the whole upgrade process works the same way. the new instance is only made available after the health check has run, and then we proceed with removing a single old instance. if the new instance is failing, then the upgrade process stops, the already existing instances should be able to handle everything.